src/poprox_recommender/handler.py

```diff
@@ -17,9 +18,11 @@ def generate_recs(event, context):
 
     body = event.get("body", {})
     is_encoded = event.get("isBase64Encoded", False)
+    is_compressed = event.get("headers", {}).get("Content-Encoding") == "gzip"
     pipeline_params = event.get("queryStringParameters", {})
 
     body = base64.b64decode(body) if is_encoded else body
+    body = gzip.decompress(body) if is_compressed else body
```

Comment:
I had a double-take on whether these operations are in the correct order because an HTTP body with content-encoding `gzip` should have gzip outside of whatever other encoding happens, but if I understand correctly the base64 encoding is when AWS base64-encodes our HTTP body before delivering it to us, we decode to get the original body and decompress to get the content.

If that is correct, would a comment about what's happening here prevent the next reader from being confused as well?

Reply:
That's correct, and I'm happy to add a comment.